### ISSUE TITLE
Make ring-jetty work

### ISSUE THREAD

**Issue by danielcompton** [BrunoBonacci/graalvm-clojure#2]
On master:

```
$ native-image --version
GraalVM Version 19.2.1 CE
$ java -version
openjdk version "1.8.0_232"
OpenJDK Runtime Environment (build 1.8.0_232-20191009173705.graal.jdk8u-src-tar-gz-b07)
OpenJDK 64-Bit GraalVM CE 19.2.1 (build 25.232-b07-jvmci-19.2-b03, mixed mode)
$ lein do clean, uberjar, native, run-native
Compiling simple.main
2019-10-24 14:38:17.909:INFO::main: Logging initialized @1477ms to org.eclipse.jetty.util.log.StdErrLog
Created /Users/danielcompton/Developer/clojure/test/graalvm-clojure/ring-jetty/target/ring-jetty-0.1.0-SNAPSHOT.jar
Created /Users/danielcompton/Developer/clojure/test/graalvm-clojure/ring-jetty/target/simple-main.jar
[./target/ring-jetty:62615]    classlist:   2,375.62 ms
[./target/ring-jetty:62615]        (cap):   1,128.28 ms
[./target/ring-jetty:62615]        setup:   2,440.11 ms
2019-10-24 14:38:26.546:INFO::ForkJoinPool-2-worker-12: Logging initialized @5707ms to org.eclipse.jetty.util.log.StdErrLog
[./target/ring-jetty:62615]   (typeflow):   5,140.84 ms
[./target/ring-jetty:62615]    (objects):   4,779.68 ms
[./target/ring-jetty:62615]   (features):     208.57 ms
[./target/ring-jetty:62615]     analysis:  10,351.47 ms
[./target/ring-jetty:62615]     (clinit):     206.89 ms
[./target/ring-jetty:62615]     universe:     503.52 ms
[./target/ring-jetty:62615]      (parse):     434.64 ms
[./target/ring-jetty:62615]     (inline):   1,418.28 ms
[./target/ring-jetty:62615]    (compile):   4,724.07 ms
[./target/ring-jetty:62615]      compile:   7,120.33 ms
[./target/ring-jetty:62615]        image:     850.46 ms
[./target/ring-jetty:62615]        write:     279.14 ms
[./target/ring-jetty:62615]      [total]:  24,094.02 ms
server started on: http://localhost:3000/
Exception in thread "main" java.lang.ClassNotFoundException: org.eclipse.jetty.server.ServerConnector
	at com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:60)
	at java.lang.Class.forName(DynamicHub.java:1174)
	at clojure.lang.RT.classForName(RT.java:2207)
	at clojure.lang.RT.classForName(RT.java:2216)
	at ring.adapter.jetty$server_connector.invokeStatic(jetty.clj:44)
	at ring.adapter.jetty$server_connector.doInvoke(jetty.clj:44)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at ring.adapter.jetty$http_connector.invokeStatic(jetty.clj:57)
	at ring.adapter.jetty$http_connector.invoke(jetty.clj:55)
	at ring.adapter.jetty$create_server.invokeStatic(jetty.clj:122)
	at ring.adapter.jetty$create_server.invoke(jetty.clj:119)
	at ring.adapter.jetty$run_jetty.invokeStatic(jetty.clj:164)
	at ring.adapter.jetty$run_jetty.invoke(jetty.clj:127)
	at simple.main$_main.invokeStatic(main.clj:15)
	at simple.main$_main.invoke(main.clj:13)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at simple.main.main(Unknown Source)
```

After applying https://github.com/ring-clojure/ring/pull/381 locally, I was able to run this. Once that is merged, we should update the dependencies and I think ring-jetty will run.

**Comment by mateusz-fiolka** [comment]
fyi: it was merged

**Comment by BrunoBonacci** [comment]
Yes, it was merged, but not released yet.

**Comment by danielcompton** [comment]
This has been released but there are other changes in the release (I think?) which are stopping me from running a native-image. I need to debug this more.

**Comment by BrunoBonacci** [comment]
I tried to build it using GraalVM 20 and Ring 1.8.0.
The compilation is successful however it fails at Runtime with the following Reflection error:

```
2020-02-22 17:42:56.571:WARN:oejs.HttpChannel:qtp2084389003-28: /
java.lang.IllegalArgumentException: No matching field found: getDispatcherType for class org.eclipse.jetty.server.Request
	at clojure.lang.Reflector.getInstanceField(Reflector.java:397)
	at clojure.lang.Reflector.invokeNoArgInstanceMember(Reflector.java:440)
	at ring.adapter.jetty$proxy_handler$fn__410.invoke(jetty.clj:25)
	at ring.adapter.jetty.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:500)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:386)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:562)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:378)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:270)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:388)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:527)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
```

more investigation is required.

**Comment by BrunoBonacci** [comment]
this is fixed via #19
